torch/csrc/jit/autodiff.cpp

```diff
@@ -0,0 +1,26 @@
+#include "torch/csrc/jit/autodiff.h"
+
+#include "torch/csrc/jit/symbolic_variable.h"
+#include "torch/csrc/utils/functional.h"
+
+namespace torch { namespace jit {
+
+std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_values) {
+  const auto build_sym_grad = [node](const std::vector<SymbolicVariable>& grads) -> std::vector<SymbolicVariable> {
+    switch(node->kind()) {
+      case kadd:
+        return {grads[0], grads[0]};
+      case ksub:
+        return {grads[0], -grads[0]};
+      case kmul:
+        auto inputs = node->inputs();
```

Comment:
nit: if we add this line before the switch, then any op that needs the inputs won't have to repeat this line, we could do the same for outputs as well. That way, we can keep many ops 1 liners.